app/src/main/java/org/fossasia/phimpme/share/flickr/FlickrActivity.java

```diff
@@ -128,52 +127,58 @@ public void onOAuthDone(OAuth result) {
-		SharedPreferences settings = getSharedPreferences(PREFS_NAME,
-				Context.MODE_PRIVATE);
-		String oauthTokenString = settings.getString(KEY_OAUTH_TOKEN, null);
-		String tokenSecret = settings.getString(KEY_TOKEN_SECRET, null);
-		if (oauthTokenString == null && tokenSecret == null) {
+		Realm realm = Realm.getDefaultInstance();
+		AccountDatabase account;
+		RealmQuery<AccountDatabase> query = realm.where(AccountDatabase.class);
+		query.equalTo("name", FLICKR.toString());
+		RealmResults<AccountDatabase> result = query.findAll();
+		if (result.size() != 0) {
+			account = result.get(0);
+			String oauthTokenString = account.getToken();
+			String tokenSecret = account.getTokenSecret();
+
+			if (oauthTokenString == null && tokenSecret == null) {
+				return null;
+			}
+			OAuth oauth = new OAuth();
+			String userName = account.getUsername();
+			String userId = account.getUserId();
+			if (userId != null) {
+				User user = new User();
+				user.setUsername(userName);
+				user.setId(userId);
+				oauth.setUser(user);
+			}
+			OAuthToken oauthToken = new OAuthToken();
+			oauth.setToken(oauthToken);
+			oauthToken.setOauthToken(oauthTokenString);
+			oauthToken.setOauthTokenSecret(tokenSecret);
+			return oauth;
+		}else
 			return null;
-		}
-		OAuth oauth = new OAuth();
-		String userName = settings.getString(KEY_USER_NAME, null);
-		String userId = settings.getString(KEY_USER_ID, null);
-		if (userId != null) {
-			User user = new User();
-			user.setUsername(userName);
-			user.setId(userId);
-			oauth.setUser(user);
-		}
-		OAuthToken oauthToken = new OAuthToken();
-		oauth.setToken(oauthToken);
-		oauthToken.setOauthToken(oauthTokenString);
-		oauthToken.setOauthTokenSecret(tokenSecret);
-		return oauth;
 	}
 
 	public void saveOAuthToken(String userName, String userId, String token, String tokenSecret) {
-
-		SharedPreferences sp = getSharedPreferences(PREFS_NAME,
-				Context.MODE_PRIVATE);
-		Editor editor = sp.edit();
-		editor.putString(KEY_OAUTH_TOKEN, token);
-		editor.putString(KEY_TOKEN_SECRET, tokenSecret);
-		editor.putString(KEY_USER_NAME, userName);
-		editor.putString(KEY_USER_ID, userId);
-		editor.apply();
-	}
-
-	public static void setInputStream(InputStream inputStream){
-		photoStream = inputStream;
-	}
-
-	public static InputStream getInputStream(){
-		return photoStream;
-	}
-
-	public static void setFilename(String file){
-		filename = file;
+		Realm realm = Realm.getDefaultInstance();
+		RealmQuery<AccountDatabase> query = realm.where(AccountDatabase.class);
+		query.equalTo("name", FLICKR.toString());
+		RealmResults<AccountDatabase> result = query.findAll();
+		AccountDatabase account;
+		realm.beginTransaction();
+		if( result.size() == 0 ) {
+			account = realm.createObject(AccountDatabase.class,FLICKR.toString());
+			account.setToken(token);
+			account.setTokenSecret(tokenSecret);
+			account.setUsername(userName);
+			account.setUserId(userId);
+		} else {
+			account = result.first();
```

Comment:
What you understood is wrong. I accept that I didnot write anything for logging out. but here it is completely for logging in. In the first call to db secret token gets saved and other elements will be empty. In the second call other details gets updated.